scipy/optimize/_minimize.py

```diff
@@ -597,53 +597,89 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
                       bounds=bounds, constraints=constraints,
                       callback=callback, **options)
 
+    if constraints is not None:
+        constraints = standardize_constraints(constraints, x0, meth)
+
+    remove_vars = False
     if bounds is not None:
         bounds = standardize_bounds(bounds, x0, meth)
 
-    if constraints is not None:
-        constraints = standardize_constraints(constraints, x0, meth)
+        lb, ub = _split_bounds(bounds)
+        btol = 1e-8  # tolerance for bounds being considered equal. Expose?
+        i_fixed = np.abs(ub - lb) < btol
+        remove_vars = i_fixed.any() and meth in {'powell', 'l-bfgs-b', 'slsqp'}
+
+    if remove_vars:
+        x_fixed = ((lb+ub)/2)[i_fixed]  # values of fixed variables
+        x0 = x0[~i_fixed]   # remove fixed variables from x0
+        bounds = _remove_from_bounds(bounds, i_fixed)
+        fun = _remove_from_func(fun, i_fixed, x_fixed)
+        if callable(jac):
+            jac = _remove_from_func(jac, i_fixed, x_fixed, remove=1)
+        if callable(hess):
+            hess = _remove_from_func(hess, i_fixed, x_fixed, remove=2)
```

Comment:
These aren't needed because we're only changing behavior when using numerical differentiation, right?

Reply:
SLSQP and LBFGSB don't use `hess`, either via FD or callable. Therefore if we're only dealing with those methods then there's no need to do tackle `hess`.
In fact `func, jac, hess` *only need to be factored if there is something downstream (func-->jac-->hess) that uses FD*. For example, if SLSQP uses FD for jac, then `x` needs to be factored out for `func` and `jac`. If SLSQP has a callable `jac` then *no* factoring needs to be done.

- check if hess is FD, if so then factorise jac and func. (doesn't need to be done for SLSQP and LBFGSB because they don't use hess).
- check if jac is FD, if so then factorise func.

```
if remove_vars:
     if is_fd(jac):
           # need to factorise func and jac
     if is_fd(hess)
           # need to factorise func, jac, hess
```

Reply:
I think we have the right logic now. Basically, if any finite differences are going to be needed - in the gradient, hessian, or constraint jacobian - and if there are fixed variables, then remove the fixed variables.